networks:
  k6:
  grafana:

services:
  influxdb:
    image: influxdb:latest
    restart: on-failure
    depends_on:
      - grafana
    networks:
      - k6
      - grafana
    ports:
      - "8086:8086"
    environment:
      - DOCKER_INFLUXDB_INIT_MODE=setup
      - DOCKER_INFLUXDB_INIT_USERNAME=admin
      - DOCKER_INFLUXDB_INIT_PASSWORD=password123
      - DOCKER_INFLUXDB_INIT_ORG=k6org
      - DOCKER_INFLUXDB_INIT_BUCKET=k6bucket
      - DOCKER_INFLUXDB_INIT_ADMIN_TOKEN=mysupersecrettoken


  grafana:
    image: grafana/grafana:latest
    networks:
      - grafana
    ports:
      - "3000:3000"
    environment:
      - GF_AUTH_ANONYMOUS_ORG_ROLE=Admin
      - GF_AUTH_ANONYMOUS_ENABLED=true
      - GF_AUTH_BASIC_ENABLED=false
      - GF_SERVER_SERVE_FROM_SUB_PATH=true
    volumes:
      - ./dashboards:/var/lib/grafana/dashboards
      - ./grafana-dashboard.yaml:/etc/grafana/provisioning/dashboards/dashboard.yaml
      - ./grafana-datasource.yaml:/etc/grafana/provisioning/datasources/datasource.yaml
    expose:
      - "3000"

volumes:
  influxdb-data:
